lib/network_utils.pm

```diff
@@ -30,7 +30,15 @@ sub setup_static_network {
     # Set default values
     $args{ip} ||= '10.0.2.15';
     $args{gw} ||= testapi::host_ip();
-    my $dns_ip = get_required_var('STATIC_DNS_SERVER');
+    #getting proper DNS from worker's config
+    open my $fh, '<', '/etc/resolv.conf' or die "error opening /etc/resolv.conf: $!";
+    my $data = do { local $/; <$fh> };
+    if($data =~ /^nameserver (.*)$/m) {
+        my $dns_ip = $1;
+    }
+    else {
+        die "Failed to get DNS ip address from worker";
+    }
```

Comment:
You can have multiple entries of nameservers. I guess this does not work as expected. Also it reads a bit complicated. Also, I doubt this can work with defining the scope of the variable `$dns_ip` just within the if.

How about just:

```perl
my $dns_ip = qx{grep -m 1 -oP '(?<=nameserver ).*' /etc/resolv.conf};
```

Reply:
just for pure perl  :)
```
my @nameserver = $data =~ (/nameserver\s+(.*)$/gm);
```

Reply:
I would say that in this case why we not just simply copy whole file ?